misk/src/main/kotlin/misk/web/HttpCall.kt

```diff
@@ -11,13 +11,31 @@ import okhttp3.RequestBody
 import okio.BufferedSink
 import okio.BufferedSource
 
+/**
+ * Information about the socket on which a HTTP call arrived.
+ */
+data class SourceSocket (
+  val type: Type,
+  val name: String
+)
+{
+  /**
+   * The type of socket on which a request arrived.
+   */
+  enum class Type {
+    NETWORK,
+    UNIX
+  }
+}
+
 /**
  * A live HTTP call from a client for use by a chain of network interceptors.
  */
 interface HttpCall {
 
   /** Immutable information about the incoming HTTP request. */
   val url: HttpUrl
+  val sourceSocket: SourceSocket?
```

Comment:
rename the field to `linkLayerAddress`. “Source” suggests origination to me, and I think this is much weaker than that.

Can you also document a disclaimer that this is the raw address known to our service, but it is very unlikely to be the originating caller’s host address? Perhaps say that this is the nearest middlebox address? (I’d be very sad if somebody did risk analysis based on these addresses.)

For follow up it’d be rad to include an originAddress field, perhaps populated by something that understands the corresponding HTTP headers in a service’s deployment topology.